src/use-swr.ts

```diff
@@ -118,11 +118,13 @@ const mutate: mutateInterface = async (
     if (error) throw error
     return data
   }
-
+  // update cached data and error, avoid notifying from the cache
   if (typeof data !== 'undefined') {
-    // update cached data, avoid notifying from the cache
     cache.set(key, data, false)
   }
+  if (typeof error !== 'undefined') {
+    cache.set(keyErr, error, false)
+  }
```

Comment:
I merged #480. I think here the error cache should be cleared even `error` is undefined.

Can you resolve the conflicts so we can merge this PR? Thanks for adding a test case! 😃

Reply:
rebased the upstream and resolved the conflicts! 🙌

Reply:
Thanks!!